### PR TITLE
api: upgrade python packages

### PR DESCRIPTION
**Pull request by JenySadadia**
Depends on https://github.com/kernelci/kernelci-core/pull/2710

Enable support for `pydantic v2` along with the latest `fastapi-pagination` package.
To enable the upgrade `fastapi` and `fastapi-users` packages are also required to be upgraded.

### CODE REVIEW

**Comment by JenySadadia** [comment]
Looks like `/doc` endpoint broke with the upgrade.
Receiving `Failed to load API definition` for https://staging.kernelci.org:9000/docs.

**Comment by pawiecz** [comment]
Regression tracker on staging seems to be working as expected with no hiccups:
```
$ docker logs kernelci-pipeline-regression_tracker
10/23/2024 08:39:14 AM UTC [INFO] Tracking regressions... 
10/23/2024 08:39:14 AM UTC [INFO] Press Ctrl-C to stop.
10/23/2024 10:49:01 AM UTC [INFO] Detected regression for node id: 6718c120e004de9d9a419d5c
10/23/2024 10:49:02 AM UTC [INFO] Regression submitted: 6718d49ee004de9d9a41b404
10/23/2024 10:49:15 AM UTC [INFO] Detected regression for node id: 6718c120e004de9d9a419d5d
10/23/2024 10:49:16 AM UTC [INFO] Regression submitted: 6718d4ace004de9d9a41b405
10/23/2024 10:51:20 AM UTC [INFO] Detected regression for node id: 6718c20fe004de9d9a41a3f9
10/23/2024 10:51:21 AM UTC [INFO] Regression submitted: 6718d529e004de9d9a41b406
10/23/2024 10:51:33 AM UTC [INFO] Detected regression for node id: 6718c20fe004de9d9a41a3fa
10/23/2024 10:51:34 AM UTC [INFO] Regression submitted: 6718d536e004de9d9a41b407
10/23/2024 11:15:55 AM UTC [INFO] Detected regression for node id: 6718c0d8e004de9d9a4199bd
10/23/2024 11:15:55 AM UTC [INFO] Regression submitted: 6718daebe004de9d9a41b408
10/23/2024 11:16:09 AM UTC [INFO] Detected regression for node id: 6718c0d8e004de9d9a4199be
10/23/2024 11:16:10 AM UTC [INFO] Regression submitted: 6718dafae004de9d9a41b409
```

File: api/models.py [hunk 1]
```diff
@@ -27,7 +29,7 @@
     Document,
     PydanticObjectId,
 )
-from bson import ObjectId
+# from bson import ObjectId
```
Comment (nuclearcat) [inline api/models.py]: Maybe we can drop this comment?

Comment (JenySadadia) [reply api/models.py]: Yup, already fixed it.

**Pull request merged**